private/bufpkg/bufmodule/bufmoduleref/bufmoduleref.go

```diff
@@ -69,6 +70,24 @@ type FileInfo interface {
 	isFileInfo()
 }
 
+// DigestChangedError is returned if module pin digests have changed unexpectedly.
+type DigestChangedError struct {
+	// CurrentDigest is the digest found in the buf.lock file.
+	CurrentDigest string
+	// UpdatedPin is a module pin with a different digest than CurrentDigest for the same commit.
+	UpdatedPin ModulePin
+}
+
+func (e *DigestChangedError) Error() string {
+	return fmt.Sprintf(
+		"module %s commit %q digest changed unexpectedly: expected=%q, found=%q",
```

Comment:
It works nice!

```
➜  moneyapis git:(main) buflon mod update
Failure: module bufbuild.internal/jfigueroa/coinbase commit "574005a7f6404734860b372d4c7667e3" digest changed unexpectedly: expected="shake256:6c661888789858cb84eeac84bb86f958e9c14f6fbdb4ea38b94fc0a2e4f04d69b45da2981d204d1baf54f8e4257a7661e8a168e04e15e2d0ca03fb49425bbe5b", found="shake256:02468788b6c17d1824c6c9759b6f5f057fec736f1fb69e13d8802ca09e5c7935a70d3d8953cade9601d4b154318172d6327e6095d342faa7e4ec1089c04476f2"
```

Since no change is being done now to see in git diff what's the issue, what about this tweak?

```suggestion
		"module %s remote commit %q returned an unexpected digest: local buf.lock=%q, returned=%q",
```

Or something in those lines...

Reply:
Updated - thanks!